api/app/Exceptions/User/UserInvalidPasswordException.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Exceptions\User;
+
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Throwable;
+
+final class UserInvalidPasswordException extends ModelNotFoundException
+{
+    public function __construct($message = "Invalid old password has been provided", $code = 401, Throwable $previous = null)
+    {
+        parent::__construct($message, $code, $previous);
+    }
+}
```

Comment:
I am not sure code `401` is appropriate here because the request is authenticated, but the entered data is wrong. Also, the base exception is incorrect. For custom exceptions like this, you need to use `BaseException`, check the `ExampleException`.